### PR TITLE
http api: makes sure header is sent even when r is not ready yet. fixes #3304

### LINKED ISSUE #3304: API call to pubsub/sub does not return HTTP header until first message arrives
This way calling `"go-ipfs-api".Request.Send()` blocks until the first message is received. Which is not nice.

### PR DESCRIPTION
**Pull request by keks**
Otherwise API clients block on pubsub subscriptions until the first packet is sent (see, well, #3304).

### CODE REVIEW

File: commands/http/handler.go [hunk 1]
```diff
@@ -298,6 +298,9 @@ func flushCopy(w io.Writer, r io.Reader) error {
 		return err
 	}
 	for {
+		// flush to send header when r is not ready yet
+		f.Flush()
```
Comment (whyrusleeping) [inline commands/http/handler.go]: probably want to check and handle the error here, just in case

**Comment by keks** [comment]
I now check the error. Also I removed the flush at the bottom of the loop so it doesn't flush pointlessly.
Now it also flushes when the reader is drained (`err, n = r.Read();  err == io.EOF && n <= 0`)

**Comment by whyrusleeping** [comment]
Tests are failing now... i'm guessing removing that final flush wasnt quite right

File: commands/http/handler.go (hunk 1 above)
Comment (keks) [inline commands/http/handler.go]: Flush doens't return an error, that's why the tests failed :D

**Comment by Kubuxu** [comment]
Can you squash it, we don't want to have any commits with build/test failures as it makes git bisecting much harder.

**Comment by keks** [comment]
@Kubuxu uh, sorry. next time!

**Pull request merged**